app/graph/types/bot_user_type.rb

```diff
@@ -1,38 +1,54 @@
-BotUserType = GraphqlCrudOperations.define_default_type do
-  name 'BotUser'
-  description 'Bot User type'
-
-  interfaces [NodeIdentification.interface]
-
-  field :avatar, types.String
-  field :name, types.String
-  field :get_description, types.String
-  field :get_version, types.String
-  field :get_source_code_url, types.String
-  field :get_role, types.String
-  field :identifier, types.String
-  field :login, types.String
-  field :dbid, types.Int
-  field :installed, types.Boolean
-  field :installations_count, types.Int
-  field :settings_ui_schema, types.String
-  field :installation, TeamBotInstallationType
-  field :default, types.Boolean
-
-  field :settings_as_json_schema do
-    type types.String
-    argument :team_slug, types.String # Some settings options are team-specific
-
-    resolve -> (bot, args, _ctx) {
-      bot.settings_as_json_schema(false, args['team_slug'])
-    }
+class BotUserType < DefaultObject
+  description "Bot User type"
+
+  implements NodeIdentification.interface
+
+  field :avatar, String, null: true
+  field :name, String, null: true
+  field :get_description, String, null: true
+
+  def get_description
+    object.get_description
+  end
+
+  field :get_version, String, null: true
+
+  def get_version
+    object.get_version
+  end
+
+  field :get_source_code_url, String, null: true
+
+  def get_source_code_url
+    object.get_source_code_url
+  end
+
+  field :get_role, String, null: true
+
+  def get_role
+    object.get_role
+  end
+
+  field :identifier, String, null: true
+  field :login, String, null: true
+  field :dbid, Integer, null: true
+  field :installed, Boolean, null: true
+  field :installations_count, Integer, null: true
+  field :settings_ui_schema, String, null: true
+  field :installation, TeamBotInstallationType, null: true
+  field :default, Boolean, null: true
+
+  field :settings_as_json_schema, String, null: true do
+    argument :team_slug, String, required: false # Some settings options are team-specific
```

Comment:
TODO: Verify input field settings (snakecaseing) carries over into subinputs

Reply:
they don't because these are arguments, not fields - we have to declare manually